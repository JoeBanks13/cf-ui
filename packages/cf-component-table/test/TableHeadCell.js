import React from 'react';
import renderer from 'react-test-renderer';
import {
  TableHeadCell,
  createTableHeadCell
} from '../../cf-component-table/src/index';
import { felaSnapshot } from 'cf-style-provider';

describe('TableHeadCell', () => {
  it('should render', () => {
    const snapshot = felaSnapshot(<TableHeadCell>TableHeadCell</TableHeadCell>);
    expect(snapshot).toMatchSnapshot();
  });

  it('should compose with styles overrides', () => {
    const TableHeadCell = createTableHeadCell(({ theme }) => ({
      verticalAlign: 'right',
      [`@media (min-width: ${theme.breakpoints.desktopLarge})`]: {
        width: '1000px'
      }
    }));
    const snapshot = felaSnapshot(<TableHeadCell>TableHeadCell</TableHeadCell>);
    expect(snapshot).toMatchSnapshot();
  });
});
