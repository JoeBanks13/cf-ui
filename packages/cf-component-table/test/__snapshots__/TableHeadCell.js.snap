// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadCell should compose with styles overrides 1`] = `
Object {
  "component": <th
    className="6zfuo5"
>
    TableHeadCell
</th>,
  "styles": ".6zfuo5:first-letter {
  text-transform: capitalize
}

.6zfuo5 {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: right;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-color: #dedede;
  text-align: left;
  font-weight: 600
}@media (max-width: 47.2em){.6zfuo5 {
  display: block;
  width: 100%;
  box-sizing: border-box;
  border-top-width: 0px
}}@media screen and (max-width: 47.2em){.6zfuo5 {
  display: block;
  border: none;
  float: left;
  clear: left;
  padding: 0px
}}@media (min-width: 97.6em){.6zfuo5 {
  width: 1000px
}}",
}
`;

exports[`TableHeadCell should render 1`] = `
Object {
  "component": <th
    className="1u7kqkt"
>
    TableHeadCell
</th>,
  "styles": ".1u7kqkt:first-letter {
  text-transform: capitalize
}

.1u7kqkt {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: middle;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-color: #dedede;
  text-align: left;
  font-weight: 600
}@media (max-width: 47.2em){.1u7kqkt {
  display: block;
  width: 100%;
  box-sizing: border-box;
  border-top-width: 0px
}}@media screen and (max-width: 47.2em){.1u7kqkt {
  display: block;
  border: none;
  float: left;
  clear: left;
  padding: 0px
}}",
}
`;
